CHANGELOG.md

```diff
@@ -17,6 +17,7 @@ IMPROVEMENTS:
 
 BUG FIXES:
  * agent: Only allow querying Prometheus formatted metrics if Prometheus is enabled within the config [[GH-10140](https://github.com/hashicorp/nomad/pull/10140)]
+ * api: Fixed race condition around headers in API client [[GH-10302](https://github.com/hashicorp/nomad/issues/10302)]
```

Comment:
I feel a bit odd about the wording here. Race condition is too low level and doesn't clarify the scope/impact.  I'd probably suggest a more specific phrasing akin: "Fixed a panic that may occur if when multiple goroutines initiate requests using the same api client"?